src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -54,20 +57,23 @@
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "The id of the network imported"),
             @ApiResponse(code = 409, message = "The study already exist or the case doesn't exists")})
-    public ResponseEntity<Void> createStudyFromExistingCase(@PathVariable("studyName") String studyName,
-                                                                  @PathVariable("caseUuid") UUID caseUuid,
-                                                                  @RequestParam("description") String description) {
-
-        if (studyService.studyExists(studyName)) {
-            throw new ResponseStatusException(HttpStatus.CONFLICT, STUDY_ALREADY_EXISTS);
-        }
-
-        if (!studyService.caseExists(caseUuid)) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, CASE_DOESNT_EXISTS);
-        }
-
-        studyService.createStudy(studyName, caseUuid, description);
-        return ResponseEntity.ok().build();
+    public Mono<ResponseEntity<Object>> createStudyFromExistingCase(@PathVariable("studyName") String studyName,
+                                                    @PathVariable("caseUuid") UUID caseUuid,
+                                                    @RequestParam("description") String description) {
+        Mono<Boolean> studyExists = studyService.studyExists(studyName);
+        Mono<Boolean> caseExists = studyService.caseExists(caseUuid);
+
+        return Mono.zip(studyExists, caseExists)
```

Comment:
did you choose on purpose to do the studyExists and caseExists calls in parallel ? Before that it was sequential.